### PR TITLE
Ci master

### PR DESCRIPTION
**Pull request by orliac**
Hi Emma,

Here is an initial version of the Jenkins CI. I squashed all the commits in branch ci-master and added an initial documentation written in Markdown in `jenkins/monitoring.md`. If people agree with the current layout I will add a HOWTO to describe how to add new stages in the monitoring.

A review from your side would be especially appreciated on the modified core files to make sure nothing will be broken when merging into `master`. You can get a list of the `A`dded and `M`odified files between `ci-master` and `master` branches with:
```
$ git diff --name-status master..ci-master
...
M       pypeline/phased_array/bluebild/data_processor.py
M       pypeline/phased_array/bluebild/field_synthesizer/fourier_domain.py
M       pypeline/phased_array/bluebild/field_synthesizer/spatial_domain.py
M       pypeline/phased_array/bluebild/imager/spatial_domain.py
...
```

After the merge, people should be submitted PRs to merge their work into branch `ci-master`. Then one needs to investigate potential warnings (and potentially the profilers' output) before we can safely merge into `master`.

As we started a new Jenkins history with the new branch `ci-master` there is currently only 2 builds. So do not hesitate to test the pipeline by pushing something in.

And of course I'm happy to iterate based on your comments and suggestions.

### CODE REVIEW

File: pypeline/phased_array/bluebild/field_synthesizer/fourier_domain.py [hunk 1]
```diff
@@ -29,6 +32,33 @@
...
+def are_close(a, b, label):
+    try:
+        print(f"are all_close {label}?", np.allclose(a, b, atol=1e-12))
+    except:
+        print(f"are all_close {label}? WENT WRONG - Check input")
+
+def check_allclose(a_, b_, label):
+    a = a_
+    b = b_
+    if cp.get_array_module(a) == cp:
```
Comment (etolley) [inline pypeline/phased_array/bluebild/field_synthesizer/fourier_domain.py:50]: isn't cp here out of scope?

File: pypeline/phased_array/bluebild/field_synthesizer/spatial_domain.py [hunk 2]
```diff
@@ -29,6 +32,60 @@ def _have_matching_shapes(V, XYZ, W):
     return False
 
 
+def dgemm_(A, B, a = 1, b = 0):
```
Comment (etolley) [inline pypeline/phased_array/bluebild/field_synthesizer/spatial_domain.py:35]: do we want to move these dgemm calls to a separate module? ie pypeline/linalg or somesuch

**Review by etolley — COMMENTED** [review]
Hi Etienne,

Thanks a lot for the extensive CPU/GPU interoperability, it all looks clean & good to me. aside from specific comments I've left in the code, I have general comments:

1) By eye the code looks fine, but did you check that the output hasn't changed?
2) Are the examples/simulation/*.py codes that have been modified being used for the ci scripts? Scripts used to run ci should probably be moved into their own ci/ or jenkins/ dir so that they can be kept relatively static.
3) should jenkins/LBN3c.png and jenkins/runtimes_all.png  be untracked?
4) what's the difference between install and new_install, requirements and new_requirements?

**Comment by orliac** [comment]
As discussed I have rearranged parts of the code and reverted others to strictly stick to the actual needs of our Jenkins pipeline and not pollute more than necessary outside of /jenkins

**Pull request closed without merging**